web/modules/custom/nys_calendar/src/Plugin/views/filter/YearFilter.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\nys_calendar\Plugin\views\filter;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\views\Plugin\views\filter\FilterPluginBase;
+
+/**
+ * Year filter.
+ *
+ * @ingroup views_filter_handlers
+ *
+ * @ViewsFilter("year_filter")
+ */
+class YearFilter extends FilterPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function valueForm(&$form, FormStateInterface $form_state) {
+    $timezone = new \DateTimeZone('America/New_York');
```

Comment:
I believe `TimeZoneResolver` service would be of use here.  Don't like to see hard-coded config data points.

Reply:
Since we're working at year granularity, is dealing with the timezone really necessary?  I understand this is the "proper" way, but it seems like overkill.

Edit: I see later on this is translating to a `BETWEEN` predicate on a timestamp field.  I guess that makes sense.